### ISSUE TITLE
[Bug]: SwatchPicker under screen reader

### ISSUE THREAD

**Issue by RongqiZ** [microsoft/fluentui#34333]
### Component

SwatchPicker

### Package version

9.58.3

### React version

17.0.1

### Environment

```shell
System:
    OS: Linux 6.8 Ubuntu 22.04.5 LTS 22.04.5 LTS (Jammy Jellyfish)
    CPU: (16) x64 AMD EPYC 7763 64-Core Processor
    Memory: 51.65 GB / 62.79 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Browsers:
    Chrome: 135.0.7049.114
  npmPackages:
    @fluentui/react: ^8.122.14 => 8.122.14 
    @fluentui/react-cards: ^0.205.203 => 0.205.203 
    @fluentui/react-tabster: ^9.23.3 => 9.23.3 
    @types/react: 17.0.45 => 17.0.45 
    @types/react-dom: 17.0.17 => 17.0.17 
    react: 17.0.1 => 17.0.1 
    react-dom: 17.0.1 => 17.0.1
```

### Current Behavior

https://react.fluentui.dev/?path=/docs/components-swatchpicker--docs
According the SwatchPicker's known limitations, The selected attribute is not supported.

When using Narrator + Edge.
Move focus into the SwatchPicker:
It does not announce grid, and not announce button, not announce selected. It may cause the user may not know the cells are actionable items. Do you have any suggested way how to use SwatchPicker to make it more accessible?

### Expected Behavior

The screen reader users knows the SwatchPicker items are actionable.

### Reproduction

https://react.fluentui.dev/?path=/docs/components-swatchpicker--docs#swatch-picker-layout

### Steps to reproduce

1. Goto the Grid section
2. Listen what narrator reads

### Are you reporting an Accessibility issue?

None

### Suggested severity

High - No workaround

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.
